pkg/sentry/fsimpl/gofer/filesystem.go

```diff
@@ -678,6 +678,16 @@ func (fs *filesystem) LinkAt(ctx context.Context, rp *vfs.ResolvingPath, vd vfs.
 		if d.nlink == math.MaxUint32 {
 			return linuxerr.EMLINK
 		}
+		if d.isSynthetic() {
+			parent.createSyntheticChildLocked(&createSyntheticOpts{
+				name:     childName,
+				mode:     mode,
+				kuid:     uid,
+				kgid:     gid,
+				endpoint: d.endpoint,
```

Comment:
initialize pipe too.
pipe: d.pipe

Reply:
ok, I'll also add corresponding syscall test for it.